### ISSUE TITLE
Chloromorphosis is not working as I thought it would be or possibly intended.

### ISSUE THREAD

**Issue by patq911** [CleverRaven/Cataclysm-DDA#23652]
Game version: Latest Experimental as of 2018/5/3

Operating system: Windows 10

Tiles or curses: Tiles

Mods active: Irrelevant.

### Expected behavior
Chloromorphosis as defined in game is ["Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have."](https://github.com/CleverRaven/Cataclysm-DDA/blob/5aeb398454f1bb415cb68718e27bd21cb0348bcd/data/json/mutations.json#L4392)

Which is defined as https://github.com/CleverRaven/Cataclysm-DDA/blob/53787ecc42c5929ffb76605fe7fbc9c4ef8ca805/src/player_hardcoded_effects.cpp#L1004

Which in turn refers to "is_in_sunlight" which is defined only as WEATHER_CLEAR or WEATHER_SUNNY https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L6352
### Actual behavior
As a result during the day, even in just cloudy overcast weather, chloromorphosis does not work at all.
### Steps to reproduce the behavior
Sleep during a cloudy/rainy day.

In my opinion, this should also add other weather types, the sun still exists even when it's cloudy. At this point this mutation isn't working as I assumed it would, personally I think this mutation should either work when you're outside and it's above a certain light level (no artificial light), whatever the light level is. Minus perhaps heavy rain, heavy snow, etc. Or the rate of hunger and thirst satiation vary based on weather. OR the wiki and description should be changed to be clear it only works in sunny and clear weather.

I believe that if you get to this point in the plant mutations tree, you kind of deserve to be given a good mutation that works as intended.

Suggested fixes: 

Add a new definition for "growable" or "acceptable" weather.

Something like: 

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L6352

`bool game::is_in_growablelight( const tripoint &p ) // Acceptable light and weather for mutation chloromorphosis
{
    return (m.is_outside( p ) && light_level( p.z ) >= 40 &&
            (weather == WEATHER_CLEAR || weather == WEATHER_SUNNY || weather == WEATHER_CLOUDY || weather == WEATHER_DRIZZLE || weather == WEATHER_RAINY || weather == WEATHER_FLURRIES));
}`

Then:
https://github.com/CleverRaven/Cataclysm-DDA/blob/53787ecc42c5929ffb76605fe7fbc9c4ef8ca805/src/player_hardcoded_effects.cpp#L1004

`        if( calendar::once_every( 10_minutes ) && has_trait( trait_id( "CHLOROMORPH" ) ) &&
            g->is_in_growablelight( pos() ) ) {
            // Hunger and thirst fall before your Chloromorphic physiology!
            if( get_hunger() >= -30 ) {
                mod_hunger( -5 );
            }
            if( get_thirst() >= -30 ) {
                mod_thirst( -5 );
            }
        }`

I'm not a programmer so I'm just guessing on some of these things here. I could start a pull request too if that makes this easier. In fact I'll do that and it can be denied if they don't like it.

edit: I don't understand markup, imagine that code being spaced correctly.

**Comment by patq911** [comment]
I've tried multiple different ways of doing this, and I just can't figure out how this works. This doesn't work for two reasons, the errors I list later, and there's some astyle error when it tries to compile with your bots. I honestly have no idea.

```cplusplus 
       if( calendar::once_every( 10_minutes ) && has_trait( trait_id( "CHLOROMORPH" ) ) &&
            g->is_outside( pos() ) && weather(WEATHER_CLEAR) || weather(WEATHER_SUNNY) ||
            weather(WEATHER_DRIZZLE) || weather(WEATHER_RAINY) || weather(WEATHER_FLURRIES)
            || weather(WEATHER_CLOUDY) || weather(WEATHER_SNOW) ) {
            // Hunger and thirst fall before your Chloromorphic physiology!
            if( get_hunger() >= -30 ) {
                mod_hunger( -5 );
            }
            if( get_thirst() >= -30 ) {
                mod_thirst( -5 );
            }
        }
```

I downloaded visual studio to stop spamming your bots. Here are the errors when I try to do this.

`1>..\src\player_hardcoded_effects.cpp(1005): error C2039: 'is_outside': is not a member of 'game'`
`1>..\src\player_hardcoded_effects.cpp(1005): error C3861: 'weather': identifier not found`

I don't understand why is_in_sunlight works but is_outside doesn't. I don't understand why weather() works in game.cpp but not here.

**Comment by Night-Pryanik** [comment]
Closed by #23775.
